vertical-pod-autoscaler/docs/flags.md

```diff
@@ -0,0 +1,89 @@
+Last updated: 2024-12-17 13:09:20 UTC
+
+| Flag | Type | Default | Description |
+|------|------|---------|-------------|
+| --address | String | :8944 | The address to expose Prometheus metrics. |
+| --client-ca-file | String | /etc/tls-certs/caCert.pem | Path to CA PEM file. |
+| --min-tls-version | String | tls1_2 | The minimum TLS version to accept.  Must be set to either tls1_2 (default) or tls1_3. |
+| --port | Int | 8000 | The port to listen on. |
+| --register-by-url | Bool | false | If set to true, admission webhook will be registered by URL (webhookAddress:webhookPort) instead of by service name |
+| --register-webhook | Bool | true | If set to true, admission webhook object will be created on start up to register with the API server. |
+| --reload-cert | Bool | false | If set to true, reload leaf certificate. |
+| --tls-cert-file | String | /etc/tls-certs/serverCert.pem | Path to server certificate PEM file. |
+| --tls-ciphers | String |  | A comma-separated or colon-separated list of ciphers to accept.  Only works when min-tls-version is set to tls1_2. |
+| --tls-private-key | String | /etc/tls-certs/serverKey.pem | Path to server certificate key PEM file. |
+| --webhook-address | String |  | Address under which webhook is registered. Used when registerByURL is set to true. |
+| --webhook-failure-policy-fail | Bool | false | If set to true, will configure the admission webhook failurePolicy to \"Fail\". Use with caution. |
+| --webhook-labels | String |  | Comma separated list of labels to add to the webhook object. Format: key1:value1,key2:value2 |
+| --webhook-port | String |  | Server Port for Webhook |
+| --webhook-service | String | vpa-webhook | Kubernetes service under which webhook is registered. Used when registerByURL is set to false. |
+| --webhook-timeout-seconds | Int | 30 | Timeout in seconds that the API server should wait for this webhook to respond before failing. |
+
+# What are the parameters to VPA recommender?
+This document is auto-generated from the flag definitions in the VPA recommender code.
+Last updated: 2024-12-17 13:09:18 UTC
+
+| Flag | Type | Default | Description |
+|------|------|---------|-------------|
+| --address | String | :8942 | The address to expose Prometheus metrics. |
+| --checkpoints-gc-interval | Duration | 10Minute | `How often orphaned checkpoints should be garbage collected` |
+| --checkpoints-timeout | Duration | 0 | `Timeout for writing checkpoints since the start of the recommender's main loop` |
+| --container-name-label | String | name | `Label name to look for container names` |
+| --container-namespace-label | String | namespace | `Label name to look for container namespaces` |
+| --container-pod-name-label | String | pod_name | `Label name to look for container pod names` |
+| --cpu-histogram-decay-half-life | Duration | 0 | `The amount of time it takes a historical CPU usage sample to lose half of its weight.` |
```

Comment:
I just realised that this default is wrong, it's actually 24 hours

Reply:
There are a few others with a default of 0, which is incorrect

Reply:
Thanks! Ill raise a PR to fix

Reply:
I see the issue: some flags are defined with constants like:
```go
DefaultOOMBumpUpRatio float64 = 1.2 // Memory is increased by 20% after an OOMKill.
```
and we don't handle them well. Do you think we should implement logic to dynamically retrieve these constants and put their values? or maybe something else?

Reply:
I'm actually not sure. If it was dynamic, that would be cool, but I'm unsure of the effort level